### PR TITLE
Fix/fuzz tests improvements

### PR DESCRIPTION
**Pull request by 0xteddybear**

### CODE REVIEW

File: test/invariants/fuzz/Protocol.t.sol [hunk 1]
```diff
@@ -14,7 +14,7 @@ contract FuzzProtocol is EchidnaTest {
   BCoWFactory factory;
   BConst bconst;
   BMath bmath;
-  BNum_exposed bnum;
```
Comment (wei3erHase) [inline test/invariants/fuzz/Protocol.t.sol:17]: why not BNum mock?

File: test/smock/MockBPool.sol [hunk 2]
```diff
@@ -53,6 +53,14 @@ contract MockBPool is BPool, Test {
...
   }
 
+  function set__reenteringMutex(bytes32 __reenteringMutex) public {
+    _reenteringMutex = __reenteringMutex;
+  }
+
+  function call__reenteringMutex() public view returns (bytes32) {
+    return _reenteringMutex;
+  }
+
```
Comment (wei3erHase) [inline test/smock/MockBPool.sol:63]: what's this? 👀

Comment (wei3erHase) [reply test/smock/MockBPool.sol:63]: ahhh because the doc had to break transient storage, ok..

Comment (0xteddybear) [reply test/smock/MockBPool.sol:63]: autogenerated getters/setters for the storage `bytes32` introduced by the doc to walk around the limitation of `hevm` not supporting transient storage

Comment (0xteddybear) [reply test/smock/MockBPool.sol:63]: this is also what breaks the integration tests btw

File: test/invariants/fuzz/Protocol.t.sol (hunk 1 above)
Comment (0xteddybear) [reply test/invariants/fuzz/Protocol.t.sol:17]: they are not generated unless I I make the functions internal

Comment (wei3erHase) [reply test/invariants/fuzz/Protocol.t.sol:17]: but we do have a MockBNum contract that externalizes all methods in `test/manual-smock/` dir

File: test/invariants/fuzz/Protocol.t.sol [hunk 3]
```diff
@@ -219,7 +220,10 @@ contract FuzzProtocol is EchidnaTest {
 
       // 19
       assert(pool.isFinalized());
-    } catch {
+    } catch (bytes memory errorData) {
+      if (keccak256(errorData) == IBPool.BPool_SpotPriceAfterBelowSpotPriceBefore.selector) {
+        assert(false);
```
Comment (wei3erHase) [inline test/invariants/fuzz/Protocol.t.sol:225]: we could add a comment saying that we're trying to hit this error but seems unreachable, or is it within `19`?

File: test/invariants/fuzz/Protocol.t.sol (hunk 1 above)
Comment (wei3erHase) [reply test/invariants/fuzz/Protocol.t.sol:17]: addressing this on #179

File: test/smock/MockBPool.sol (hunk 2 above)
Comment (wei3erHase) [reply test/smock/MockBPool.sol:63]: addressing this on #179

File: test/invariants/fuzz/BToken.t.sol [hunk 4]
```diff
@@ -1,11 +1,12 @@
+// SPDX-License-Identifier: MIT
```
Comment (wei3erHase) [inline test/invariants/fuzz/BToken.t.sol:1]: why only this one is MIT? i think we should make all test files MIT

File: test/invariants/fuzz/Protocol.t.sol [hunk 5]
```diff
@@ -554,15 +567,5 @@ contract FuzzProtocol is EchidnaTest {
...
-
-contract BNum_exposed is BNum {
-  function bdiv_exposed(uint256 a, uint256 b) public pure returns (uint256) {
-    return bdiv(a, b);
-  }
-
-  function bmul_exposed(uint256 a, uint256 b) public pure returns (uint256) {
-    return bmul(a, b);
-  }
+  // function fuzz_settlerSettle() public {}
```
Comment (wei3erHase) [inline test/invariants/fuzz/Protocol.t.sol:570]: should we remove this alltogether?

File: test/invariants/PROPERTIES.md [hunk 6]
```diff
@@ -88,4 +89,4 @@ power of a power should mult the exp (x^a)^b == x^(a*b)
 ## Untested (precision issues in test settingsq)
 calcOutGivenIn should be inv with calcInGivenOut
 calcPoolOutGivenSingleIn should be inv with calcSingleInGivenPoolOut
-calcSingleOutGivenPoolIn should be inv with calcPoolInGivenSingleOut
\ No newline at end of file
+calcSingleOutGivenPoolIn should be inv with calcPoolInGivenSingleOut
```
Comment (wei3erHase) [inline test/invariants/PROPERTIES.md:92]: weird..

File: test/invariants/fuzz/Protocol.t.sol [hunk 7]
```diff
@@ -227,6 +229,7 @@ contract FuzzProtocol is EchidnaTest {
       // 19
       assert(pool.isFinalized());
     } catch (bytes memory errorData) {
+      // 25
```
Comment (wei3erHase) [inline test/invariants/fuzz/Protocol.t.sol:232]: 👌

**Pull request merged**